cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -215,6 +214,32 @@ func ReadStaticPodFromDisk(manifestPath string) (*v1.Pod, error) {
 	return pod, nil
 }
 
+// GetEtcdProbeURL will return a validated --advertise-client-url URL or https://127.0.0.1:2379
+// to use for etcd liveness probes. in static pod manifests.
+func GetEtcdProbeURL(cfg *kubeadmapi.MasterConfiguration, componentName string) string {
+	if cfg.Etcd.ExtraArgs != nil {
+		if arg, exists := cfg.Etcd.ExtraArgs[etcdAdvertiseClientURLsArg]; exists {
+			// basic etcd url validation. Reference:
+			// https://github.com/coreos/etcd/blob/master/pkg/types/urls.go#L35-L47
```

Comment:
Thanks for providing this link.
Would you please pin it to the commit?
https://github.com/coreos/etcd/blob/54d56e2531a3d3fff0f80a15371cc18a68b83e52/pkg/types/urls.go#L35-L47